src/netstring_protocol.cpp

```diff
@@ -172,3 +172,14 @@ size_t netstring_client_t::stream_responses(const void* message, size_t size, bo
 }
 
 } // namespace prime_server
+
+prime_server::shortcircuiter_t<prime_server::netstring_entity_t> make_shortcircuiter(const std::string& health_check_str){
+  return [health_check_str](const prime_server::netstring_entity_t& request) {
+    if(request.body == health_check_str){
+      netstring_entity_t response = netstring_entity_t::from_string("healthy", 7);
+      std::string response_str = response.to_string();
+      return std::unique_ptr<zmq::message_t>(new zmq::message_t(response_str.size(), response_str.c_str()));
```

Comment:
im having second thoughts about using `unique_ptr` as the return type here. we really should construct the response once and just use it any time the healthcheck message is needed rather than construct a new one every time with the same contents every time.

perhaps a cleaner implementation would be to, outside of the lambda, create a `shared_ptr` and then capture that by copy `[=]` and just return the shared ptr if the string matches. the same would happen in the http protocol too except there we need to maybe make some of the response information on the fly potentially

Reply:
This is gold, i would have never thought about the overhead to recreate this objects every time! I think i could address it!